Source/Utils/Parser/IntervalsParser.cpp

```diff
@@ -193,43 +193,51 @@ utils::parser::BTDIntervalsParser::BTDIntervalsParser (
         const int start = temp_slice.getStart();
         const int period = temp_slice.getPeriod();
         int btd_iter_ind;
-        // for Slice temp_slice in m_slices,
-        // determine the index in m_btd_iterations where temp_slice's starting value goes
-        //
         // Implementation note:
-        // assuming the user mostly lists slices in ascending order,
         // start at the end of m_btd_iterations and search backward
+        // with the thinking that the user mostly lists slices in ascending order
+        //
+        // for Slice temp_slice in m_slices,
+        // determine the largest index in m_btd_iterations smaller than start
         if (m_btd_iterations.size() == 0)
         {
             btd_iter_ind = 0;
         }
         else
         {
             btd_iter_ind = m_btd_iterations.size() - 1;
-            while (start < m_btd_iterations[btd_iter_ind] and btd_iter_ind>0)
+            while (start < m_btd_iterations.at(btd_iter_ind) and btd_iter_ind>0)
             {
                 btd_iter_ind--;
             }
         }
         // insert each iteration contained in temp_slice into m_btd_iterations
         // adding them in increasing sorted order and not adding any iterations
         // already contained in m_btd_iterations
-        for (int ii = start; ii <= temp_slice.getStop(); ii += period)
+        for (int slice_iter = start; slice_iter <= temp_slice.getStop(); slice_iter += period)
         {
             if (m_btd_iterations.size() > 0)
             {
-                // find where iteration ii should go in m_btd_iterations
-                while (ii > m_btd_iterations[btd_iter_ind] && btd_iter_ind < static_cast<int>(m_btd_iterations.size()))
+                // increment btd_iter_ind until slice_iter <= m_btd_iterations.at(btd_iter_ind)
+                while (btd_iter_ind < static_cast<int>(m_btd_iterations.size())
+                        && slice_iter > m_btd_iterations.at(btd_iter_ind))
                 {
                     btd_iter_ind++;
                 }
-                if (ii != m_btd_iterations[btd_iter_ind])
+                // this is the place to insert slice_iter if it is not in m_btd_iterations
+                // if slice_iter > all entries in m_btd_iterations, append slice_iter
+                if (btd_iter_ind == static_cast<int>(m_btd_iterations.size()))
```

Comment:
```suggestion
                if (btd_iter_ind == num_btw_iterations)
```

Reply:
One could argue that the old style is better because one does not need to go back to find out what `num_btw_iterations` is to know what this test is about.

Reply:
I don't think so with a proper name: is only few lines up and increases readability by naming things properly.
(Also `.size()` on vectors is more costly than querying a member.)